src/GovUk.Education.ExploreEducationStatistics.Admin/Services/FootnoteService.cs

```diff
@@ -60,83 +56,70 @@ public class FootnoteService : IFootnoteService
             return await _contentPersistenceHelper
                 .CheckEntityExists<Release>(releaseId)
                 .OnSuccess(_userService.CheckCanUpdateRelease)
-                .OnSuccess(async () =>
-                {
-                    var newFootnoteId = _guidGenerator.NewGuid();
-
-                    var footnote = new Footnote
-                    {
-                        Id = newFootnoteId,
-                        Content = content,
-                        Subjects = CreateSubjectLinks(newFootnoteId, subjectIds),
-                        Filters = CreateFilterLinks(newFootnoteId, filterIds),
-                        FilterGroups = CreateFilterGroupLinks(newFootnoteId, filterGroupIds),
-                        FilterItems = CreateFilterItemLinks(newFootnoteId, filterItemIds),
-                        Indicators = CreateIndicatorsLinks(newFootnoteId, indicatorIds)
-                    };
-
-                    await _context.Footnote.AddAsync(footnote);
-
-                    await _context.ReleaseFootnote.AddAsync(new ReleaseFootnote
-                    {
-                        ReleaseId = releaseId,
-                        Footnote = footnote
-                    });
-
-                    await _context.SaveChangesAsync();
-                    return footnote;
-                })
+                .OnSuccess(() => _footnoteRepository.CreateFootnote(
+                    releaseId,
+                    content,
+                    filterIds,
+                    filterGroupIds,
+                    filterItemIds,
+                    indicatorIds,
+                    subjectIds))
```

Comment:
Would suggest using named parameters here given that we have so many lists of ids. We wouldn't want to accidentally assign the wrong one if the parameter order changes.

Reply:
Have made this change now as part of the new commit altering from lists to sets, and done it in quite a few other places, thanks.